include/openthread-types.h

```diff
@@ -37,13 +37,19 @@
 
 #include <stdint.h>
 #include <stdbool.h>
+
 #include <platform/toolchain.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
+ * This type represents the OpenThread instance structure.
+ */
+typedef struct otInstance otInstance;
```

Comment:
I've done this trick a few times myself. It's rather convenient since it allows you to avoid casting.